webapp/src/app/admin/teams/table/teams-table.component.ts

```diff
@@ -0,0 +1,241 @@
+    const nameFilter = this._nameFilter()?.trim()?.toLowerCase();
+    if (nameFilter && nameFilter.length > 0) {
+      return this._teams().filter((u) => u.name!.toLowerCase().includes(nameFilter));
+    }
+    return this._teams();
+  });
+  private readonly _nameSort = signal<'ASC' | 'DESC' | null>(null);
+  protected readonly _filteredSortedPaginatedTeams = computed(() => {
+    const sort = this._nameSort();
+    const start = this._displayedIndices().start;
+    const end = this._displayedIndices().end + 1;
+    const names = this._filteredNames();
+    if (!sort) {
+      return names.slice(start, end);
+    }
+    return [...names].sort((p1, p2) => (sort === 'ASC' ? 1 : -1) * p1.name.localeCompare(p2.name)).slice(start, end);
+  });
+  protected readonly _allFilteredPaginatedTeamsSelected = computed(() => this._filteredSortedPaginatedTeams().every((team) => this._selected().includes(team)));
+  protected readonly _checkboxState = computed(() => {
+    const noneSelected = this._selected().length === 0;
+    const allSelectedOrIndeterminate = this._allFilteredPaginatedTeamsSelected() ? true : 'indeterminate';
+    return noneSelected ? false : allSelectedOrIndeterminate;
+  });
+
+  protected readonly _trackBy: TrackByFunction<TeamInfo> = (_: number, u: TeamInfo) => u.id;
+  protected readonly _totalElements = computed(() => this._filteredNames().length);
+  protected readonly _onStateChange = ({ startIndex, endIndex }: PaginatorState) => this._displayedIndices.set({ start: startIndex, end: endIndex });
+
+  constructor() {
+    // needed to sync the debounced filter to the name filter, but being able to override the
+    // filter when loading new users without debounce
+    effect(() => this._nameFilter.set(this._debouncedFilter() ?? ''), { allowSignalWrites: true });
+  }
+
+  protected toggleTeam(team: TeamInfo) {
+    this._selectionModel.toggle(team);
+  }
+
+  protected handleHeaderCheckboxChange() {
+    const previousCbState = this._checkboxState();
+    if (previousCbState === 'indeterminate' || !previousCbState) {
+      this._selectionModel.select(...this._filteredSortedPaginatedTeams());
+    } else {
+      this._selectionModel.deselect(...this._filteredSortedPaginatedTeams());
+    }
+  }
+
+  protected handleNameSortChange() {
+    const sort = this._nameSort();
+    if (sort === 'ASC') {
+      this._nameSort.set('DESC');
+    } else if (sort === 'DESC') {
+      this._nameSort.set(null);
+    } else {
+      this._nameSort.set('ASC');
+    }
+  }
+
+  protected deleteTeam(team: TeamInfo) {
+    if (this.isLoading()) {
+      return;
+    }
+    this.adminService.deleteTeam(team.id!).subscribe({
+      next: () => this.invalidateTeams(),
+      error: (err) => console.error('Error deleting team', err)
+    });
+  }
+
+  protected copyName(element: TeamInfo) {
+    console.log('Copying name', element);
+    navigator.clipboard.writeText(element.name!);
+  }
+
+  _newLabelName = new FormControl('');
```

Comment:
Move this up, can you also group the properties with some comments since you have so many